tests/compiler/nonnullable.ts

```diff
@@ -0,0 +1,13 @@
+
+
+type nullableString = string | null;;
+
+assert(nameof<NonNullable<u32>>() == nameof<u32>());
+assert(nameof<NonNullable<nullableString>>() == nameof<string>());
+assert(nameof<NonNullable<string[] | null>>() == nameof<string[]>());
+
+function assertNonNull<T>(): void {
+  assert(!isNullable<T>(), "T cannot be null");
+}
+
+assertNonNull<NonNullable<nullableString>>();
```

Comment:
wait. It should be a compile time error, not runtime error. Isn't?

Reply:
Only if `assertNonNull` would take an argument, which evaluates to a type that is nullable contrary to the type argument which is not, I think. As-is, it is only calling a function with some type., right?

Reply:
Yes, so I would like to see this test with real argument (not type params).

Reply:
Yes, this code should be optimized away.  But the compiler error is still tested in the other tests.  Above I'm using string compare, which does require the binary to compile.